x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -686,6 +686,20 @@ private static void checkFilterOnAggs(LogicalPlan p, Set<Failure> localFailures,
                         }
                     }
                 });
+            } else {
+                Set<Expression> unsupported = new LinkedHashSet<>();
+                filter.condition().forEachDown(Expression.class, e -> {
+                    Expression f = attributeRefs.resolve(e, e);
+                    if (f instanceof TopHits) {
+                        unsupported.add(f);
+                    }
+                });
+                if (unsupported.isEmpty() == false) {
+                    String plural = unsupported.size() > 1 ? "s" : StringUtils.EMPTY;
+                    localFailures.add(
+                            fail(filter.condition(), "HAVING filter is unsupported for function" + plural + " {}",
```

Comment:
I have mixed feelings about the error message. For `SELECT * FROM (SELECT first(salary) AS f FROM test_emp) WHERE f > 0` there is no `HAVING` in the query, but the error message says there is one. This can be confusing to the user, who doesn't know that we transform the `WHERE` into a `HAVING` because this is how an aggregation is being filtered.

How about `filtering is unsupported for function [first(salary)]` for that query?